mcs/build/corcompare.make

```diff
@@ -1,4 +1,4 @@
-API_INFO = $(MONO_PATH) $(RUNTIME) $(topdir)/class/lib/$(PROFILE)/mono-api-info.exe
+API_INFO = $(MONO_PATH) $(RUNTIME) $(topdir)/class/lib/$(PROFILE)$(if $(PROFILE_PLATFORM),-$(PROFILE_PLATFORM))/mono-api-info.exe
```

Comment:
I don't like this, we need to introduce LIB_PROFILE or something like that instead of copy and paste everywhere